src/python/pants/backend/tools/semgrep/subsystem.py

```diff
@@ -51,6 +57,20 @@ class SemgrepSubsystem(PythonToolBase):
     register_lockfile = True
     default_lockfile_resource = ("pants.backend.tools.semgrep", "semgrep.lock")
 
+    config_paths = StrListOption(
+        default=[".semgrep"],
+        help=softwrap(
+            """
+            The locations of semgrep configs
+            """
+        ),
+    )
+
+    ignore_config_path = StrOption(
+        default=".semgrepignore",
```

Comment:
Ah... sorry about forcing you to re-investigate everything about these ignore files. This was known in the original addition of the backend, and cut out of scope: https://github.com/pantsbuild/pants/pull/18593#discussion_r1184441437

I even referenced https://github.com/huonw/pants/pull/1 "so I don't forget about it", but look how far that's gone. 🙈 

In any case, I think these two things are true, and still true as of this PR:

- pants runs `semgrep` in the sandbox root (i.e. working directory is the root of the repository)
- semgrep only looks for `.semgrepignore` in its working directory

Given this, running semgrep via pants will only ever successfully read `//.semgrepignore`. Thus, someone who sets `ignore_config_path = "some/sub/directory/.semgrepignore` will just have semgrep not using the ignore. It'll be included in the sandbox, but not used!

(The thinking behind the globbing behaviour was "future proofing": if a new version of semgrep started reading the nested `.semgrepignore`s, it'll automatically work with Pants, because Pants is including them in the sandboxes.)

---

Two questions:

1. Are you thinking of changing the old behaviour (including all files called `.semgrepignore`) because it was causing problems?
2. Do you have an example of how someone would customise this path (and why)?

Reply:
Ah gotcha, I'm on my phone right now but I'll look at those PRs later.

No, it wasn't causing problems. Basically when I was adding the description for the semgrepignore customization, I initially had a note that it was hierarchical. That got me curious, so I tried to use it, and realized it did not work as expected. Probably should've done a git blame to see why it was done, but I decided to mention explicitly that it had to be at the root-based path (since this is different from the rules dir behavior), and remove the relevant code.

Shall I add it back with a note about context? Though I do think as it stands pants taking this on will be a huge burden in taking over tool responsibility full stuff. The simplest way I can think to get it done would be to to inject `:include`s into the root semgrepignore. Is there any precedent to alter file digests like that?

Reply:
And yeah, just realized semgrepignore is not actually customizable, so should just remove this and go back to the hard-coded value.

Reply:
😂 hilarious that we ended up with nearly the same integration tests for showing this behavior

Reply:
removed this configurability

Reply:
> Though I do think as it stands pants taking this on will be a huge burden in taking over tool responsibility stuff. The simplest way I can think to get it done would be to to inject :includes into the root semgrepignore during the pants run. Is there any precedent to alter file digests like that?

Yes, I agree that Pants should try to avoid having too much "smarts" tailored to each tool.

My thinking with the continuing work related to `.semgrepignore` (i.e. https://github.com/huonw/pants/pull/1) was just to make it more obvious when something is ignored, to reduce confusion. Not "invent" a new file.